trade-finance-manager-ui/scripts/moj-filter-hide.ts

```diff
@@ -0,0 +1,52 @@
+import $ from 'jquery';
+
+declare global {
+  interface Window {
+    MOJFrontend: any;
+  }
+}
+
+type FilterVisibility = 'SHOWN' | 'HIDDEN';
+
+const filterSelector = '.moj-filter-layout__filter';
+const toggleButtonContainerSelector = '.moj-action-bar__filter';
+const toggleButtonContainer = document.querySelector(toggleButtonContainerSelector)!;
+
+const params = (toggleButtonContainer as HTMLElement).dataset;
+const FILTER_VISIBILITY_SESSION_KEY = `${params.filteredTableId}-filters-visibility-state`;
+
+const setFilterVisibility = (visibility: FilterVisibility) => {
+  sessionStorage.setItem(FILTER_VISIBILITY_SESSION_KEY, visibility);
+};
+
+const cachedShownStatus = sessionStorage.getItem(FILTER_VISIBILITY_SESSION_KEY) as FilterVisibility | null;
```

Comment:
Are we able to make something optional rather than `| null`?

Reply:
`getItem` returns `string | null` - I think it's good to apply the type casting straight away and also keep the consistency with the underlying API. Not sure how I feel about changing the type casting to `FilterVisibility | undefined`.